assets/tests/api.test.ts

```diff
@@ -1122,3 +1125,33 @@ describe("putRouteTabs", () => {
     expect(args!.body).toEqual(JSON.stringify({ route_tabs: routeTabs }))
   })
 })
+
+describe("putDetourUpdate", () => {
+  const testToggleMachine = createMachine({
+    id: "testToggle",
+    initial: "inactive",
+    states: {
+      inactive: {
+        on: {
+          TOGGLE: "active",
+        },
+      },
+      active: {
+        on: {
+          TOGGLE: "inactive",
+        },
+      },
+    },
+  })
+  const actor = createActor(testToggleMachine).start()
+  const persistedSnapshot = actor.getPersistedSnapshot()
+
+  test("uses PUT and CSRF token", () => {
+    mockFetch(200, { data: 1 })
+    putDetourUpdate(persistedSnapshot)
+    expect(window.fetch).toHaveBeenCalledTimes(1)
+    const args = jest.mocked(window.fetch).mock.calls[0][1]
+    expect(args!.method).toEqual("PUT")
+    expect(args!.headers).toHaveProperty("x-csrf-token")
```

Comment:
question/suggestion(non-blocking): could this be written using jest asymmetric matchers?

This may be the wrong code, but something like this?
```ts
expect(jest.mocked(window.fetch))
	.toHaveBeenCalledWith([
		expect.anything(),
		expect.objectContaining({
			method: "PUT",
			headers: expect.objectContaining({
				"x-csrf-token": expect.any(String) // or expect.anything(), I dunno the right syntax for specifying the type
			})
		})
	])
```

Reply:
https://jestjs.io/docs/expect#asymmetric-matchers